db/model/Link.ts

```diff
@@ -0,0 +1,38 @@
+import {
+    Entity,
+    PrimaryGeneratedColumn,
+    Column,
+    BaseEntity,
+    ManyToOne,
+    Relation,
+} from "typeorm"
+import { Gdoc } from "./Gdoc/Gdoc.js"
+
+export const gdocUrlRegex =
+    /https:\/\/docs\.google\.com\/document\/u\/\d\/d\/([^\/]+)\/edit/
```

Comment:
There seem to be a few different google doc URL formats. I found this type of URL for example: `https://docs.google.com/document/u/0/d/1wrXxKFgkpy8E5etN9jEJOTmZKtrld9AmllNY8DQM1-g/edit` which this regex fails to parse.

I think it would be good to add a comment with example urls that this regex parses and to adapt it so that it can also parse wha I posted. Maybe a regex like this is better? `https:\/\/docs\.google\.com\/.+?\/[-\w]+/edit`